alphagov/paas-cf
# Configure buildpack versions with YAML

What
----

Updating the pipeline every time we need to update the buildpacks is not very sustainable, and is quite confusing to do without missing something.

`config/buildpacks.yml` is a configuration file that describes the state of our buildpacks, with each described as follows:

```yaml
- name: binary_buildpack
  repo_name: binary-buildpack
  stack: cflinuxfs2
  version: v1.0.27
  sha: 90e04df12df78f6d06ef127aa9992f505b7e51e21cb5ac97b68bc7e742193635
  filename: binary-buildpack-cflinuxfs2-v1.0.27.zip
  url: https://github.com/cloudfoundry/binary-buildpack/releases/download/v1.0.27/binary-buildpack-cflinuxfs2-v1.0.27.zip
  dependencies: []
```

Concourse will now run `parse_buildpacks_and_run_uploader.rb` which parses buildpacks.yml, and runs the existing `upload-buildpack.sh` script to download from GitHub and upload to cloudfoundry.

How to review
-------------

* Sanity check that all versions and SHAs are the same between the original configuration in the pipeline and `config/buildpacks.yml`
* Make a note of current buildpacks in cloudfoundry: `cf buildpacks`
* Delete the buildpacks:
    ```sh
    cf delete-buildpack -s cflinuxfs2 -f staticfile_buildpack
    cf delete-buildpack -s cflinuxfs2 -f java_buildpack
    cf delete-buildpack -s cflinuxfs2 -f ruby_buildpack
    cf delete-buildpack -s cflinuxfs2 -f dotnet_core_buildpack
    cf delete-buildpack -s cflinuxfs2 -f nodejs_buildpack
    cf delete-buildpack -s cflinuxfs2 -f go_buildpack
    cf delete-buildpack -s cflinuxfs2 -f python_buildpack
    cf delete-buildpack -s cflinuxfs2 -f php_buildpack
    cf delete-buildpack -s cflinuxfs2 -f binary_buildpack
    cf delete-buildpack -s cflinuxfs3 -f staticfile_buildpack
    cf delete-buildpack -s cflinuxfs3 -f java_buildpack
    cf delete-buildpack -s cflinuxfs3 -f ruby_buildpack
    cf delete-buildpack -s cflinuxfs3 -f dotnet_core_buildpack
    cf delete-buildpack -s cflinuxfs3 -f nodejs_buildpack
    cf delete-buildpack -s cflinuxfs3 -f go_buildpack
    cf delete-buildpack -s cflinuxfs3 -f python_buildpack
    cf delete-buildpack -s cflinuxfs3 -f php_buildpack
    cf delete-buildpack -s cflinuxfs3 -f binary_buildpack
    ```

* Deploy this branch to your environment and confirm the same versions are reinstalled

Who can review
--------------

Not @tnwhitwell or and @richardTowers
